goreq.go

```diff
@@ -30,6 +30,7 @@ type Request struct {
 	Method            string
 	Uri               string
 	Body              interface{}
+	BodyReader        io.Reader
```

Comment:
Why is this necessary if `prepareRequestBody` already returns a reader?. Can't we use that?